Modules/mmapmodule.c

```diff
@@ -291,6 +291,17 @@ mmap_read_line_method(mmap_object *self,
     return result;
 }
 
+#if defined(MS_WIN32) && !defined(DONT_USE_SEH)
```

Comment:
I forget the exact reason, but ctypes only uses SEH on 32-bit Windows. Possibly because the x64/ARM64 calling conventions don't allow recovery across separately compiled modules? Or there's a reliance on compiler-specific metadata that isn't always there? Whatever it is, that shouldn't matter for this scenario.

While MSVC is our only supported compiler that will set `MS_WINDOWS`, people sometimes use Clang to compile for Windows (and I guess GCC, but I haven't heard of anyone doing that recently). I presume those don't support `__try`, but it's probably best to omit it based on compiler detection rather than a separate flag.

(Also, `ctypes` is... ehm... not the best example of good coding conventions 😉 )

Reply:
So I just check for MS_WINDOWS and ignore MS_WIN32 and DONT_USE_SEH?

Reply:
`MS_WIN32` is always defined on Windows. Restricting to 64-bit requires checking for `MS_WIN64`. ctypes uses SEH on all supported versions of Windows as long as `DONT_USE_SEH` is not defined.